plugins/backend-trace-forwarder/src/Cardano/BM/Backend/TraceForwarder.lhs

```diff
@@ -186,51 +187,69 @@ instance Exception TraceForwarderBackendFailure
 \begin{code}
-spawnDispatcher :: ToJSON a => TraceForwarderMVar a -> IO (Async.Async ())
-spawnDispatcher tfMVar = Async.async $ processQueue
+spawnDispatcher :: ToJSON a => Configuration -> TraceForwarderMVar a -> IO (Async.Async ())
+spawnDispatcher config tfMVar = do
+  -- To reduce network traffic it's possible to send log items not one by one,
+  -- but collect them in the queue and periodically send this list as one ByteString.
+  forwardDelay <- getForwardDelay config >>= \case
+    Nothing -> pure defaultDelayInMs
+    Just delayInMs -> pure delayInMs
+  Async.async $ processQueue forwardDelay
  where
-  processQueue :: IO ()
-  processQueue = do
+  -- If the configuration doesn't specify forward delay, use default one.
+  defaultDelayInMs :: Word
+  defaultDelayInMs = 1000
+
+  processQueue :: Word -> IO ()
+  processQueue delayInMs = forever $ do
+    threadDelay $ fromIntegral delayInMs * 1000
     currentTF <- readMVar tfMVar
-    -- Read the next log item from the queue. If the queue is still empty -
-    -- blocking and waiting for the next log item.
-    nextItem <- atomically $ TBQ.readTBQueue (tfQueue currentTF)
+    itemsList <- atomically $ TBQ.flushTBQueue (tfQueue currentTF)
     -- Try to write it to the handle. If there's a problem with connection,
     -- this thread will initiate re\-establishing of the connection and
     -- will wait until it's established.
-    sendItem tfMVar nextItem
-    -- Continue...
-    processQueue
+    sendItems config tfMVar itemsList
 
--- Try to send log item to the handle.
-sendItem :: ToJSON a => TraceForwarderMVar a -> LogObject a -> IO ()
-sendItem tfMVar lo =
+-- Try to send log items to the handle.
+sendItems :: ToJSON a => Configuration -> TraceForwarderMVar a -> [LogObject a] -> IO ()
+sendItems _ _ [] = return ()
+sendItems config tfMVar items@(lo:_) =
   tfHandle <$> readMVar tfMVar >>= \case
     Nothing -> do
       -- There's no handle, initiate the connection.
       establishConnection 1 1 tfMVar
       -- Connection is re\-established, try to send log item.
-      sendItem tfMVar lo
+      sendItems config tfMVar items
     Just h ->
       try (BSC.hPutStrLn h $! encodedHostname) >>= \case
-        Right _ ->
+        Right _ -> do
           -- Hostname was written to the handler successfully,
-          -- try to write serialized LogObject.
-          try (BSC.hPutStrLn h $! bs) >>= \case
+          -- try to write serialized list of LogObjects.
+          -- To reduce network traffic, we can use compression before sending.
+          --
+          -- TODO: Currently compressing mechanism is disabled, because there's
+          -- some problems with decompression (in TraceAcceptor).
+          -- Will be fixed in the future.
+          bsToSend <- getForwardGZip config >>= \case
+                        Nothing      -> pure bs
+                        Just _isGZip -> pure bs
+          try (BSC.hPutStrLn h $! bsToSend) >>= \case
```

Comment:
bsToSend can be commented out including the call into the configuration.
```suggestion
          try (BSC.hPutStrLn h $! bs) >>= \case
```

Reply:
Fixed.